core/src/gossiper.rs

```diff
@@ -0,0 +1,167 @@
+    /// Limits that all transactions need to obey, in terms of size
+    /// of WASM blob and number of instructions.
+    transaction_limits: TransactionLimits,
+}
+
+impl Gossiper {
+    /// Start [`Self`] actor.
+    pub fn start(self) -> GossiperHandle {
+        let (message_sender, message_receiver) = mpsc::channel(1);
+        tokio::task::spawn(self.run(message_receiver));
+        GossiperHandle { message_sender }
+    }
+
+    /// Construct [`Self`] from configuration
+    pub fn from_configuration(
+        // Currently we are using configuration parameters from sumeragi not to break configuration
+        configuartion: &Configuration,
+        network: IrohaNetwork,
+        queue: Arc<Queue>,
+        sumeragi: Arc<Sumeragi>,
+    ) -> Self {
+        Self {
+            queue,
+            sumeragi,
+            network,
+            transaction_limits: configuartion.transaction_limits,
+            gossip_batch_size: configuartion.gossip_batch_size,
+            gossip_period: Duration::from_millis(configuartion.gossip_period_ms),
+        }
+    }
+
+    async fn run(self, mut message_receiver: mpsc::Receiver<TransactionGossip>) {
+        let mut gossip_period = tokio::time::interval(self.gossip_period);
+        #[allow(clippy::arithmetic_side_effects)]
+        loop {
+            tokio::select! {
+                _ = gossip_period.tick() => self.gossip_transactions(),
+                transaction_gossip = message_receiver.recv() => {
+                    let Some(transaction_gossip) = transaction_gossip else {
+                        iroha_logger::info!("All handler to Gossiper are dropped. Shutting down...");
+                        break;
+                    };
+                    self.handle_transaction_gossip(transaction_gossip);
+                }
+            }
+            tokio::task::yield_now().await;
+        }
+    }
+
+    fn gossip_transactions(&self) {
+        let txs = self
+            .queue
+            .n_random_transactions(self.gossip_batch_size, &self.sumeragi.wsv_mutex_access());
+
+        if txs.is_empty() {
+            iroha_logger::debug!("Nothing to gossip");
+            return;
+        }
+
+        iroha_logger::trace!(tx_count = txs.len(), "Gossiping transactions");
+        self.network.broadcast(Broadcast {
+            data: NetworkMessage::Gossiper(Box::new(TransactionGossip::new(txs))),
+        });
+    }
+
+    fn handle_transaction_gossip(&self, TransactionGossip { txs }: TransactionGossip) {
+        iroha_logger::trace!(size = txs.len(), "Received new transaction gossip");
+        for tx in txs {
+            enqueue_transaction(
+                tx,
+                &self.queue,
+                &self.sumeragi.wsv_mutex_access(),
+                &self.transaction_limits,
+            );
+        }
+    }
+}
+
+fn enqueue_transaction(
+    tx: VersionedSignedTransaction,
+    queue: &Queue,
+    wsv: &WorldStateView,
+    transaction_limits: &TransactionLimits,
+) {
```

Comment:
```suggestion
fn enqueue_transaction(&self, tx) {
```

I would make it a method